patt/src/main/java/org/example/strategy/Navigator.java

```diff
@@ -0,0 +1,14 @@
+package org.example.strategy;
+public class Navigator {
```

Comment:
Navigator should have a field of RouteStrategy routeStrategy.
and calculateRoute() method that accept destination and use this routeStrategy to calculate route (and you can set different routes) or pass via constructior needed route strategy -- newBusRouteStrategy, etc